Packs/ArcherRSA/Integrations/ArcherV2/ArcherV2.py

```diff
@@ -0,0 +1,987 @@
+
+    errors = get_errors_from_res(res)
+    if errors:
+        return_error(errors)
+
+    if res.get('RequestedObject') and res.get('IsSuccessful'):
+        attachment_id = res['RequestedObject'].get('Id')
+        return_outputs(f'File uploaded succsessfully, attachment ID: {attachment_id}', {}, res)
+    else:
+        return_error('Upload file failed')
+
+
+def download_file_command(client: Client, args: Dict[str, str]):
+    attachment_id = args.get('attachment-id')
+    res = client.do_request('GET', f'rsaarcher/api/core/content/attachment/{attachment_id}')
+
+    errors = get_errors_from_res(res)
+    if errors:
+        return_error(errors)
+
+    if res.get('RequestedObject') and res.get('IsSuccessful'):
+        content = base64.b64decode(res['RequestedObject'].get('AttachmentBytes'))
+        filename = res['RequestedObject'].get('AttachmentName')
+        return demisto.results(fileResult(filename, content))
+    else:
+        return_error('File downloading failed', outputs=res)
+
+
+def list_users_command(client: Client, args: Dict[str, str]):
+    user_id = args.get('user-id')
+    if user_id:
+        res = client.do_request('GET', f'rsaarcher/api/core/system/user/{user_id}')
+    else:
+        res = client.do_request('GET', f'rsaarcher/api/core/system/user')
+
+    errors = get_errors_from_res(res)
+    if errors:
+        return_error(errors)
+
+    if isinstance(res, dict):
+        res = [res]
+
+    users = []
+    for user in res:
+        if user.get('RequestedObject') and user.get('IsSuccessful'):
+            user_obj = user['RequestedObject']
+            users.append({'Id': user_obj.get('Id'),
+                          'DisplayName': user_obj.get('DisplayName'),
+                          'FirstName': user_obj.get('FirstName'),
+                          'MiddleName': user_obj.get('MiddleName'),
+                          'LastName': user_obj.get('LastName'),
+                          'AccountStatus': ACCOUNT_STATUS_DICT[user_obj.get('AccountStatus')],
+                          'LastLoginDate': user_obj.get('LastLoginDate'),
+                          'UserName': user_obj.get('UserName')})
+
+    markdown = tableToMarkdown('Users list', users)
+    context: dict = {
+        f'Archer.User(val.Id && val.Id == obj.Id)':
+            users
+    }
+    return_outputs(markdown, context, res)
+
+
+def search_records_command(client: Client, args: Dict[str, str]):
+    app_id = args.get('application-id')
+    field_to_search = args.get('field-to-search-on')
+    search_value = args.get('search-value')
+    max_results = args.get('max-results', 10)
+    date_operator = args.get('date-operator')
+    numeric_operator = args.get('numeric-operator')
+    fields_to_display = argToList(args.get('fields-to-display'))
+    fields_to_get = argToList(args.get('fields-to-get'))
+    full_data = args.get('full-data')
+    full_data = True if (full_data and full_data.lower() == 'true') else False
```

Comment:
better to handle the arg in the following (assuming `true` is the default value):
`full_data = args.get('full-data', 'true') == 'true'`